src/masternode.cpp

```diff
@@ -491,31 +414,26 @@ bool CMasternodeBroadcast::Create(CTxIn txin, CService service, CKey keyCollater
              CBitcoinAddress(pubKeyCollateralAddressNew.GetID()).ToString(),
              pubKeyMasternodeNew.GetID().ToString());
 
-
-    CMasternodePing mnp(txin);
-    if(!mnp.Sign(keyMasternodeNew, pubKeyMasternodeNew)) {
-        strErrorRet = strprintf("Failed to sign ping, masternode=%s", txin.prevout.ToStringShort());
+    auto Log = [&strErrorRet,&mnbRet](std::string sErr)->bool
+    {
+        strErrorRet = sErr;
         LogPrintf("CMasternodeBroadcast::Create -- %s\n", strErrorRet);
         mnbRet = CMasternodeBroadcast();
         return false;
-    }
+    };
+
+    CMasternodePing mnp(txin);
+    if (!mnp.Sign(keyMasternodeNew, pubKeyMasternodeNew))
+        return Log(strprintf("Failed to sign ping, masternode=%s", txin.prevout.ToStringShort()));
 
     mnbRet = CMasternodeBroadcast(service, txin, pubKeyCollateralAddressNew, pubKeyMasternodeNew, PROTOCOL_VERSION);
 
-    if(!mnbRet.IsValidNetAddr()) {
-        strErrorRet = strprintf("Invalid IP address, masternode=%s", txin.prevout.ToStringShort());
-        LogPrintf("CMasternodeBroadcast::Create -- %s\n", strErrorRet);
-        mnbRet = CMasternodeBroadcast();
-        return false;
-    }
+    if (!mnbRet.IsValidNetAddr())
+        return Log(strprintf("Invalid IP address, masternode=%s", txin.prevout.ToStringShort()));
 
     mnbRet.lastPing = mnp;
-    if(!mnbRet.Sign(keyCollateralAddressNew)) {
-        strErrorRet = strprintf("Failed to sign broadcast, masternode=%s", txin.prevout.ToStringShort());
-        LogPrintf("CMasternodeBroadcast::Create -- %s\n", strErrorRet);
-        mnbRet = CMasternodeBroadcast();
-        return false;
-    }
+    if (!mnbRet.Sign(keyCollateralAddressNew))
+        return Log(strprintf("Failed to sign broadcast, masternode=%s", txin.prevout.ToStringShort()));
```

Comment:
We have lost mnbRet invalidation here. I think,  it can be dangerous (it's enough to call Relay on this invalid object to get problems). Current use cases process this situation correctly with Create's returned result,  but in the future it can be the reason of some errors.